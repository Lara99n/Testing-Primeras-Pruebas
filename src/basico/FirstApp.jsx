import React from "react";
import PropTypes from "prop-types";

export const App = () => {
  const app = "lara";

  return <p>{app}</p>;
};

export default function FirstApp({ titulo }) {
  return (
    <>
      <h2>{App()}</h2>
      <h3>FirstApp</h3>
      <h3>{titulo}</h3>
    </>
  );
}

FirstApp.propTypes = {
  titulo: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  titulo: "No hay titulo",
};
