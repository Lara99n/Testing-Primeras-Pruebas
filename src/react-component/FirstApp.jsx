import React from "react";
import PropTypes from "prop-types";

export default function FirstApp({ titulo, subtitulo, name }) {
  return (
    <>
      <h1 data-testid="test-title">{titulo}</h1>
      <p>{subtitulo}</p>
      <p>{subtitulo}</p>
      <p>{name}</p>
    </>
  );
}

FirstApp.propTypes = {
  titulo: PropTypes.string.isRequired,
  subtitulo: PropTypes.string,
};

FirstApp.defaultProps = {
  name: "Lara Albornoz",
  subtitulo: "No hay subtitulo",
  /*  titulo: "No hay titulo", */
};
