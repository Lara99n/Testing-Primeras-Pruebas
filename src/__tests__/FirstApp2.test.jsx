import { render, screen } from "@testing-library/react";
import FirstApp from "../react-component/FirstApp";

describe("Pruebas en FirstApp", () => {
  const titulo = "Hola, soy Goku";
  let subTitulo = "soy un subtitulo";

  test("debe de hacer match con el snaptshot", () => {
    const { container } = render(<FirstApp titulo={titulo} />);
    expect(container).toMatchSnapshot();
  });

  test('debe demostrar el mensaje "Hola sot Goku"', () => {
    render(<FirstApp titulo={titulo} />);
    expect(screen.getByText(titulo)).toBeTruthy();
  });

  test("debe de mostrar el titulo en un h1", () => {
    render(<FirstApp titulo={titulo} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      titulo
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp titulo={titulo} subTitulo={subTitulo} />);

    /*  expect(screen.getAllByText(subTitulo).length).toBe(2); */
  });
});
