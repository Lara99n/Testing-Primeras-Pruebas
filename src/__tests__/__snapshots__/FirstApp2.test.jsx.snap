// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe de hacer match con el snaptshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Lara Albornoz
  </p>
</div>
`;
